core/arch/arm/kernel/thread.c

```diff
@@ -253,6 +257,10 @@ void thread_set_exceptions(uint32_t exceptions)
 {
 	uint32_t daif = read_daif();
 
+	/* IRQ must not be unmasked while holding a spinlock */
+	if (!(exceptions & THREAD_EXCP_IRQ))
+		assert_have_no_spinlock();
```

Comment:
If we already have a spinlock then IRQ should already be masked or some other logic has failed.
But regardless of that, this will never make the situation worse.